### ISSUE TITLE
ProfilePlanarBase - [merged]

### ISSUE THREAD

**Issue by hejamu** [maicos-devel/maicos#222]
In GitLab by @PicoCentauri on Apr 4, 2022, 13:33 UTC

_Merges DensityPlanarBase -> main_

This MR introduces a new child class for the PlanarBase class: `ProfilePlanarBase`. This class allows calculating 
atomistic profiles by providing a suitable function. `ProfilePlanarBase` allows for different normalizations like 
'number' used for the temperature and in the future velocity profiles as well as 'volume' used for densities.

Following the new class I split the former huge and complicated 
`DensityPlanar` class into three seperate classes: `DensityPlanar`, `ChemicalPotentialPlanar` and 
`TemperaturePlanar`. With this the code becomes cleaner and also easier for testing and maintaining.

PR Checklist
------------
 - [x] Physical Testing
 - [x] Update Docs
 - [x] Update Tests

*Migrated from GitLab: https://gitlab.com/maicos-devel/maicos/-/merge_requests/110*

**Comment by hejamu** [comment]
In GitLab by @codecov on Apr 4, 2022, 13:37 UTC

# [Codecov](https://codecov.io/gl/maicos-devel/MAICoS/pull/110?src=pr&el=h1&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS) Report
> Merging [#110](https://codecov.io/gl/maicos-devel/MAICoS/pull/110?src=pr&el=desc&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS) (b4934a9) into [main](https://codecov.io/gl/maicos-devel/MAICoS/commit/14193ac44f3aed05d2e7ae80f6ad3390e19a6055?el=desc&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS) (14193ac) will **increase** coverage by `1.36%`.
> The diff coverage is `91.11%`.

```diff
@@            Coverage Diff             @@
##             main     #110      +/-   ##
==========================================
+ Coverage   79.03%   80.40%   +1.36%     
==========================================
  Files          12       12              
  Lines        1326     1342      +16     
  Branches      212      203       -9     
==========================================
+ Hits         1048     1079      +31     
+ Misses        214      205       -9     
+ Partials       64       58       -6     
```


| [Impacted Files](https://codecov.io/gl/maicos-devel/MAICoS/pull/110?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS) | Coverage Δ | |
|---|---|---|
| [maicos/modules/density.py](https://codecov.io/gl/maicos-devel/MAICoS/pull/110/diff?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS#diff-bWFpY29zL21vZHVsZXMvZGVuc2l0eS5weQ==) | `87.89% <ø> (+0.71%)` | :arrow_up: |
| [maicos/modules/structure.py](https://codecov.io/gl/maicos-devel/MAICoS/pull/110/diff?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS#diff-bWFpY29zL21vZHVsZXMvc3RydWN0dXJlLnB5) | `51.34% <0.00%> (+0.19%)` | :arrow_up: |
| [maicos/utils.py](https://codecov.io/gl/maicos-devel/MAICoS/pull/110/diff?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS#diff-bWFpY29zL3V0aWxzLnB5) | `59.13% <ø> (+3.35%)` | :arrow_up: |
| [maicos/modules/base.py](https://codecov.io/gl/maicos-devel/MAICoS/pull/110/diff?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS#diff-bWFpY29zL21vZHVsZXMvYmFzZS5weQ==) | `94.11% <95.52%> (+0.87%)` | :arrow_up: |
| [maicos/\_\_init\_\_.py](https://codecov.io/gl/maicos-devel/MAICoS/pull/110/diff?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS#diff-bWFpY29zL19faW5pdF9fLnB5) | `88.88% <100.00%> (ø)` | |
| [maicos/decorators.py](https://codecov.io/gl/maicos-devel/MAICoS/pull/110/diff?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS#diff-bWFpY29zL2RlY29yYXRvcnMucHk=) | `96.87% <100.00%> (+5.80%)` | :arrow_up: |
| [maicos/modules/epsilon.py](https://codecov.io/gl/maicos-devel/MAICoS/pull/110/diff?src=pr&el=tree&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS#diff-bWFpY29zL21vZHVsZXMvZXBzaWxvbi5weQ==) | `86.25% <100.00%> (+0.54%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gl/maicos-devel/MAICoS/pull/110?src=pr&el=continue&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gl/maicos-devel/MAICoS/pull/110?src=pr&el=footer&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS). Last update [14193ac...b4934a9](https://codecov.io/gl/maicos-devel/MAICoS/pull/110?src=pr&el=lastupdated&utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=gitlab&utm_content=comment&utm_campaign=pr+comments&utm_term=MAICoS).

**Comment by hejamu** [comment]
LGTM, I really like this change. Just needs some coverage tests.

The ChemicalPotentialPlanar could be written even simpler now, but that will be a topic for another time.

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 8, 2022, 10:32 UTC

added 1 commit

<ul><li>99e1a936 - Address issues of rounding errors for std</li></ul>

[Compare with previous version](/maicos-devel/maicos/-/merge_requests/110/diffs?diff_id=369707664&start_sha=18001dc871b5eedf6d12a3bbcf1e9a650fc54e5e)

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 8, 2022, 11:27 UTC

added 17 commits

<ul><li>99e1a936...2b69e2d0 - 16 commits from branch <code>maicos-devel:main</code></li><li>24607231 - Merge branch &#39;main&#39; into DensityPlanarBase</li></ul>

[Compare with previous version](/maicos-devel/maicos/-/merge_requests/110/diffs?diff_id=369761302&start_sha=99e1a936716e906bc625c38c701257349e79cc4a)

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:04 UTC

added 8 commits

<ul><li>e4e5ba08...14193ac4 - 4 commits from branch <code>maicos-devel:main</code></li><li>4a488e8e - Tests for docstring and ProfilePlanarBase</li><li>6b02ff62 - Merge branch &#39;main&#39; into DensityPlanarBase</li><li>6103de94 - Fix new import</li><li>e8e05a5f - Fixed all tests and docs</li></ul>

[Compare with previous version](/maicos-devel/maicos/-/merge_requests/110/diffs?diff_id=378967408&start_sha=e4e5ba087e36922e8ece06d15573f65d3b44e83e)

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:32 UTC

requested review from @hejamu

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:38 UTC

Commented on [tests/modules/test_density.py line 154](https://github.com/maicos-devel/maicos/compare/14193ac44f3aed05d2e7ae80f6ad3390e19a6055..b4934a9a6701ae62f00bec6d18343d12dd5e45fc#diff-f148f4976a1691eb5d5399c8cba670cd18e0ae48b07311db0fa619f3d416e34fR154)



I don't why the error is now doubled but I think we can leave it for now and tackle the error estimation later.

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:40 UTC

added 1 commit

<ul><li>c5af1af8 - Minor styling and typos</li></ul>

[Compare with previous version](/maicos-devel/maicos/-/merge_requests/110/diffs?diff_id=379004344&start_sha=6290131edb9dd5077f4a674329343c9b0bfead7a)

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:41 UTC

marked the checklist item **Update Tests** as completed

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:41 UTC

marked the checklist item **Update Docs** as completed

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:41 UTC

marked the checklist item **Physical Testing** as completed

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 10:45 UTC

added 1 commit

<ul><li>b9d3d5e9 - Update README</li></ul>

[Compare with previous version](/maicos-devel/maicos/-/merge_requests/110/diffs?diff_id=379008372&start_sha=c5af1af862d269c9c5d1741f8a38a27df33611d3)

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on Apr 22, 2022, 11:00 UTC

Some tests are failing because they depend on random positions and the conversion of A^3 to nm^3 multiplies the error by a absolute difference of 1000. Once we merged !113 in main repo and afterwards here they will go away.

TL;DR: Failing tests are no bugs but will be fixed with !113.

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on May 4, 2022, 09:23 UTC

added 1 commit

<ul><li>b4934a9a - Fix seed for random numbers</li></ul>

[Compare with previous version](/maicos-devel/maicos/-/merge_requests/110/diffs?diff_id=386469434&start_sha=b9d3d5e994b525085b09c5c014daa8e43566144e)

**Comment by hejamu** [comment]
approved this merge request

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on May 6, 2022, 10:19 UTC

mentioned in commit 70ba69e965ce6985f2958d8f3a543907d938790c

**Comment by hejamu** [comment]
In GitLab by @PicoCentauri on May 6, 2022, 10:19 UTC

mentioned in commit f0d8b9cb2fe72f4e56c0624c34d1b7b1b208b274
